cmd/eksctl-anywhere/cmd/options.go

```diff
@@ -23,6 +27,17 @@ func (c clusterOptions) mountDirs() []string {
 	return dirs
 }
 
+func mountPrivateKeyFile() (string, bool) {
+	privateKeyFilePath, err := git.GetTokenFromEnv(EksaGitPrivateKeyTokenEnv)
+	if err != nil {
+		return "", false
+	}
+	if validations.FileExistsAndIsNotEmpty(privateKeyFilePath) {
```

Comment:
Shouldn't this return an error if it doesn't exist?
The method signature is a bit weird. If the file not existing is not considered an error, then can we just signify that scenario with an empty string?

Reply:
I understand the env var might not be set, but if it is, and the file doesn't exist, shouldn't we return an error? Btw, that kind of validation should probably be somewhere else

If we want to see if the env var is set, we should probably use `os.LookupEnv`

Summary of what I'm proposing:

* If the env var is not set, ignore it and don't mount anything
* If it's set, mount it and validate it is a valid and existing path (but move the validation somewhere else)

Reply:
The validation currently does exist in the preflight validations, however we mount the dirs before getting to the preflight validations, which is why we validate the file here as well.